snet_cli/commands.py

```diff
@@ -504,3 +510,247 @@ def transact(self):
             return receipt, events
         else:
             self._error("Cancelled")
+
+
+class ServiceCommand(BlockchainCommand):
+
+    # We are sorting files before we add them to the .tar since an archive containing the same files in a different
+    # order will produce a different content hash; sorting order is the same default sorting order used in GNU tar.
+    def _tar_imports(self, paths):
+        max_depth = 0
+        deepest_path = PurePath()
+        for path in paths:
+            path_depth = len(path.parts)
+            if not max_depth or path_depth < max_depth:
+                max_depth = path_depth
+                deepest_path = path
```

Comment:
this is the shallowest path, not the deepest

Reply:
Changed